backend/coins/eth/account_test.go

```diff
@@ -158,3 +159,40 @@ func TestTxProposal(t *testing.T) {
 		require.Equal(t, errors.ErrInvalidAddress, errp.Cause(err))
 	})
 }
+
+func TestMatchesAddress(t *testing.T) {
+	acct := newAccount(t)
+	defer acct.Close()
+	acct.Synchronizer.WaitSynchronized()
+
+	// Test invalid Ethereum address
+	t.Run("Invalid Ethereum address", func(t *testing.T) {
+		matches, err := acct.MatchesAddress("invalid_address")
+		require.Error(t, err)
+		require.Equal(t, matches, false)
+		assert.Contains(t, err.Error(), "invalidAddress")
```

Comment:
can check here that it is actually `errors.ErrInvalidAddress` 

    require.Equal(s.T(), errors.ErrInvalidAddress, errp.Cause(err))